gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/transactions/SnapshotTransactionDUnit.java

```diff
@@ -0,0 +1,1141 @@
+
+    VM server1 = this.serverVMs.get(0);
+    VM server2 = this.serverVMs.get(1);
+    server1.invoke(new SerializableRunnable() {
+      @Override
+      public void run() {
+        TXManagerImpl txMgr = GemFireCacheImpl.getExisting()
+            .getCacheTransactionManager();
+        txMgr.setObserver(new TransactionObserverAdapter() {
+          Object lock1 = new Object();
+          Object lock2 = new Object();
+          boolean alreadyWaiting = false;
+          @Override
+          public void duringIndividualCommit(TXStateProxy tx,
+              Object callbackArg) {
+
+            if (callbackArg == "test") {
+              synchronized (lock1) {
+                lock1.notify();
+              }
+              return;
+            }
+            if(!alreadyWaiting) {
+              alreadyWaiting = true;
+              try {
+                synchronized (lock1) {
+                  lock1.wait();
+                }
+              } catch (InterruptedException ie) {
+                // ignore
+              }
+            }
+          }
+
+          public void notifyCommit() {
+            synchronized (lock1) {
+              lock1.notify();
+            }
+          }
+        });
+      }
+    });
+
+    server2.invoke(new SerializableRunnable() {
+      @Override
+      public void run() {
+        TXManagerImpl txMgr = GemFireCacheImpl.getExisting()
+            .getCacheTransactionManager();
+        txMgr.setObserver(new TransactionObserverAdapter() {
+          Object lock1 = new Object();
+          Object lock2 = new Object();
+          boolean alreadyWaiting = false;
+          @Override
+          public void duringIndividualCommit(TXStateProxy tx,
+              Object callbackArg) {
+
+            if (callbackArg == "test") {
+              synchronized (lock1) {
+                lock1.notify();
+              }
+              return;
+            }
+            if(!alreadyWaiting) {
+              alreadyWaiting = true;
+              try {
+                synchronized (lock1) {
+                  lock1.wait();
+                }
+              } catch (InterruptedException ie) {
+                // ignore
+              }
+            }
+          }
+
+          public void notifyCommit() {
+            synchronized (lock1) {
+              lock1.notify();
+            }
+          }
+        });
+      }
+    });
+
+    Thread t = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          final Connection conn = TestUtil.getConnection();
+          Statement stmt = conn.createStatement();
+          for (int i = 0; i < 5; i++) {
+            String stmtString = "insert into " + tableName + " values(" + i + ",'test" + i + "')";
+            stmt.addBatch(stmtString);
+          }
+          stmt.executeBatch();
+        } catch (Exception e) {
+          exception[0] = e;
+          e.printStackTrace();
+        }
+      }
+    });
+    t.start();
+    Thread.sleep(1000);
```

Comment:
This time based sleep can be dangerous when running in suite or is it ok. What is the purpose of the sleep?